### PR TITLE
Feature/demo v2

### PR DESCRIPTION
**Pull request by donikv**

### CODE REVIEW

File: Demo/Viper-v2-Demo/Modules/Login/LoginViewController.swift [hunk 1]
```diff
@@ -24,7 +24,8 @@ final class LoginViewController: UIViewController {
     
     @IBOutlet private weak var loginButton: UIButton!
     @IBOutlet private weak var loginButtonBottomMargin: NSLayoutConstraint!
-
+    @IBOutlet weak var registerButton: UIButton!
```
Comment (fgulan) [inline Demo/Viper-v2-Demo/Modules/Login/LoginViewController.swift]: private outlet.

File: Demo/Viper-v2-Demo/Modules/New Group/RegisterViewController.swift [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        view.endEditing(true)
+    }
+    
+    @IBAction func closeButtonActionHandler() {
+        presenter.didSelectCloseAction()
+    }
+    
+    // MARK: Private
+    
+    private func registerForKeyboardNotifications() {
```
Comment (fgulan) [inline Demo/Viper-v2-Demo/Modules/New Group/RegisterViewController.swift]: Use underscore for private funcs,

File: Demo/Viper-v2-Demo/Modules/Login/LoginViewController.swift [hunk 3]
```diff
@@ -52,6 +53,10 @@ final class LoginViewController: UIViewController {
         )
     }
     
+    @IBAction func registerButtonActionHandler() {
```
Comment (Truba) [inline Demo/Viper-v2-Demo/Modules/Login/LoginViewController.swift]: Should be private.

File: Demo/Viper-v2-Demo/Modules/Login/LoginViewController.swift [hunk 4]
```diff
@@ -52,6 +53,10 @@ final class LoginViewController: UIViewController {
...
+        presenter.didSelectRegisterAction()
+    }
+    
     @IBAction func tapGestureRecognizerActionHandler() {
```
Comment (Truba) [inline Demo/Viper-v2-Demo/Modules/Login/LoginViewController.swift]: Should be private.

File: Demo/Viper-v2-Demo/Modules/New Group/RegisterPresenter.swift [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+    // MARK: - Private properties -
+
+    static private let minimumPasswordLenght: UInt = 6
+    
+    private unowned var _view: RegisterViewInterface
+    private var _interactor: RegisterInteractorInterface
+    private var _wireframe: RegisterWireframeInterface
+    
+    private let _authorizationManager = AuthorizationAdapter.shared
```
Comment (Truba) [inline Demo/Viper-v2-Demo/Modules/New Group/RegisterPresenter.swift]: It's a better practise to inject this things.

File: Demo/Viper-v2-Demo/Modules/New Group/RegisterViewController.swift [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    }
+    
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        NotificationCenter.default.removeObserver(self)
+    }
+    
+    // MARK: Actions
+    
+    @IBAction func registerButtonActionHandler() {
```
Comment (Truba) [inline Demo/Viper-v2-Demo/Modules/New Group/RegisterViewController.swift]: Should be private.

File: Demo/Viper-v2-Demo/Modules/New Group/RegisterViewController.swift [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+        presenter.didSelectRegisterAction(with: usernameTextField.text, email: emailTextField.text, password: passwordTextField.text, confirmedPassword: confirmPasswordTextField.text)
+    }
+    
+    @IBAction func tapGestureRecognizerActionHandler() {
```
Comment (Truba) [inline Demo/Viper-v2-Demo/Modules/New Group/RegisterViewController.swift]: Should be private.

File: Demo/Viper-v2-Demo/Modules/New Group/RegisterViewController.swift [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+        view.endEditing(true)
+    }
+    
+    @IBAction func closeButtonActionHandler() {
```
Comment (Truba) [inline Demo/Viper-v2-Demo/Modules/New Group/RegisterViewController.swift]: Should be private.

File: Demo/Viper-v2-Demo/Modules/New Group/RegisterViewController.swift [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+        )
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(keyboardWillHide(_:)),
+            name: .UIKeyboardWillHide,
+            object: nil
+        )
+    }
+    
+    @objc func keyboardWillShow(_ notification: Notification) {
```
Comment (Truba) [inline Demo/Viper-v2-Demo/Modules/New Group/RegisterViewController.swift]: Should be private.

File: Demo/Viper-v2-Demo/Modules/New Group/RegisterViewController.swift [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+        view.setNeedsUpdateConstraints()
+        
+        UIView.animate(withDuration: 0.3) {
+            self.ballImageView.alpha = 0.0
+            self.logoImageView.alpha = 0.0
+            self.view.layoutIfNeeded()
+        }
+    }
+    
+    @objc func keyboardWillHide(_ notification: Notification) {
```
Comment (Truba) [inline Demo/Viper-v2-Demo/Modules/New Group/RegisterViewController.swift]: Should be private.

**Pull request merged**